### DISCUSSION TITLE
track X,Y coordiantes

### DISCUSSION THREAD

**Discussion by w-i-z-a-r-d** [theOehrly/Fast-F1#157] [Q&A]
we can get the car's position ['X',"Y','Z'] from the telemetry, but what about the track coordinates so we can plot the track and the car's position with respect to the track.

**Comment by siddharth-tewari** [comment]
I have a code which makes the track layout photo. Hope it helps but only problem is that it uses "Session.load_laps" which will be deprecated. You need to fix it. 
It just picks up the fastest lap of the race and plot using car's position only.
I don't think track position is available. 
<b>Code</b>
```python
import fastf1 as ff1
import matplotlib.font_manager as fm
from fastf1 import plotting
from matplotlib import pyplot as plt
from labellines import labelLine, labelLines

def add_arrow(line, position=None, start_ind=None, direction='right', size=15, color=None):
   
    if color is None:
        color = line.get_color()

    xdata = line.get_xdata()
    ydata = line.get_ydata()

    if position is None:
        position = xdata.mean()
    # find closest index
    if start_ind is None:
        start_ind = len(xdata) // 2

    if direction == 'right':
        end_ind = start_ind + 1
    else:
        end_ind = start_ind - 1

    line.axes.annotate('',
        xytext=(xdata[start_ind], ydata[start_ind]),
        xy=(xdata[end_ind], ydata[end_ind]),
        arrowprops=dict(arrowstyle="-|>", color=color),
        size=size
    )

ff1.Cache.enable_cache('/cache')  # optional but recommended

year = 2022
wknd = 1
ses = 'R'

weekend = ff1.get_session(year, wknd)
session = ff1.get_session(year, wknd, ses)
laps = session.load_laps(with_telemetry=True)
lap = laps.pick_fastest()

# Track with sectors

sec1 = lap.Sector1Time
sec2 = lap.Sector1Time+lap.Sector2Time
sec3 = lap.Sector1Time+lap.Sector2Time+lap.Sector3Time
times = lap.telemetry[['Time', 'X', 'Y']]
rslt1 = times[(times['Time'] <= sec1)]
rslt2 = times[(times['Time'] <= sec2) & (times['Time'] > sec1)]
rslt3 = times[(times['Time'] <= sec3) & (times['Time'] > sec2)]

fig2, ax2 = plt.subplots(figsize=(12, 6.75))

prop = fm.FontProperties(fname='F1.otf')
fig2.suptitle(f'{weekend.name} {year}', fontproperties=prop, size=24, y=0.97)

ax2.plot(lap.telemetry['X'], lap.telemetry['Y'], color='black',  linestyle='-', linewidth=16)
line1 = ax2.plot(rslt1['X'], rslt1['Y'], label='sector 1', color='red', linestyle='-', linewidth=3)[0]
line2 = ax2.plot(rslt2['X'], rslt2['Y'], label='sector 2', color='cornflowerblue', linestyle='-', linewidth=3)[0]
line3 = ax2.plot(rslt3['X'], rslt3['Y'], label='sector 3', color='gold', linestyle='-', linewidth=3)[0]

add_arrow(line1, start_ind=1, size=20, color='white')
labelLines(plt.gca().get_lines(), backgroundcolor='black', fontsize=8, zorder=2.5)

plt.subplots_adjust(left=0.01 ,right=0.99, top=0.9, bottom=0.01)
plt.axis('off')
plt.show()
```
<b>Output</b>
![image](https://user-images.githubusercontent.com/76590952/164032714-e6debf7e-208e-4a8d-b6cc-0969539d5055.png)

**Reply by walteij** [reply]
Can you also share the:
`prop = fm.FontProperties(fname='F1.otf')` 
F1.otf file :)

**Reply by siddharth-tewari** [reply]
@walteij Here's the file link. GitHub doesn't directly support that format so used link.
https://ufile.io/hjbt2ivb

**Reply by walteij** [reply]
Great, most tracks look ok, but Silverstone and Jeddah are really wonkey.... Seems like they're rotated and then scaled differently.
Any ideas on how to fix that?
![2021_Silverstone_Circuitmap](https://user-images.githubusercontent.com/64311753/164989201-e83e15a2-edbc-4471-9207-e1dc1ed116ea.png)
![2021_Jeddah_Circuitmap](https://user-images.githubusercontent.com/64311753/164989206-a3ab8d8d-e13e-496f-914c-46ae546728b9.png)

**Reply by theOehrly** [reply]
How they're rotated is one thing. But you need to enable equal axis ratio when plotting, so that Matplotlib does not distort the image.

**Reply by walteij** [reply]
Thanks for the tip :)
So I added this code after line 73 and it looks a lot better now for Silverstone. I suppose Jeddah is a bit more difficult since it is a very stretched circuit.
`    ratio = 1
    x_left, x_right = ax2.get_xlim()  
    y_low, y_high = ax2.get_ylim()  
    ax2.set_aspect(abs((x_right-x_left)/(y_low-y_high))*ratio)`
![2021_Silverstone_Circuitmap](https://user-images.githubusercontent.com/64311753/164989983-c794cc72-23b5-40d3-b5a0-145941eb2b49.png)

**Reply by theOehrly** [reply]
There's a simpler way 

https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.axis.html

**Comment by w-i-z-a-r-d** [comment]
thanks for your reply, for this, i am able to draw the track based on X,Y,Z of the driver in any lap wanted, but I was looking to get the tracks X,Y,Z in any way/ from any API or site, so I would be able to animate the car on that track using the X,Y,Z of the car and X,Y,Z of the track.

**Reply by theOehrly** [reply]
There are no reference coordinates for the track. You'll have to generate them from the telemetry data or just use a single representative lap for it
